serve/mlc_serve/engine/engine_common.py

```diff
@@ -132,6 +134,51 @@ def detokenize_incrementally(
     return delta
 
 
+def logprob_detokenize(
+        tokenizer: TokenizerP,
+        logprob_info: Optional[RawLogprobsInfo],
+) -> Optional[LogprobsContent]:
+    """Detokenize tokens from RawLogprobInfo and convert the latter to LogprobContent"""
+    if logprob_info is None:
+        return None
+
+    top_logprobs: List[TopLogprobs] = []
+    if (
+        logprob_info.top_tokens is not None and
+        logprob_info.top_logprobs is not None
+    ):
+        top_tokens = zip(logprob_info.top_tokens[:], logprob_info.top_logprobs[:])
+        count: Dict[str, int] = {}
+        # dedup duplicates
+        # Todo: Make sure decode can generate different tokens
+        for top_token, _ in top_tokens:
+            detokenized = tokenizer.decode(top_token)
```

Comment:
I don't understand why we need to detokenize here specifically for logprob. Moreover, this is probably not the right way to detokenize (usually we need to prepend prefix tokens and use `tokenizer.convert_ids_to_tokens(...)`).

Can we somehow use properly detokenized results from https://github.com/octoml/mlc-llm/blob/batch-serving/serve/mlc_serve/engine/engine_common.py#L102, since they are already available when `logprob_detokenize` is called.